luigi/task.py

```diff
@@ -544,7 +544,7 @@ def __repr__(self):
         repr_parts = []
         param_objs = dict(params)
         for param_name, param_value in param_values:
-            if param_objs[param_name].significant:
+            if param_objs[param_name].significant and param_objs[param_name].visibility == ParameterVisibility.PUBLIC:
```

Comment:
I think this may bring up a valid question. Should `significant` parameters be allowed to be anything other than `public`?
I'm inclined to think if a parameter is defined as significant, it should be included in its task's `__repr__`.

@nryanov You added the parameter visibility feature. What are your thoughts on my comment above?

Reply:
Good question.  I'm not a heavy user of Luigi, but so far I'd be fine with a boolean on parameters to control whether they are exposed (logs, email, etc).

Reply:
Could you validate if adding `significant=False` to the Parameter instantiation correctly removes it from the error email's subject?

The [docstring for this parameter configuration](https://github.com/spotify/luigi/blob/master/luigi/parameter.py#L145-L149) continues to lead me to believe that non-public parameters should not be significant.

Reply:
Sorry for delayed answer

@dlstadther this is a good question. I think that  significant parameters may be public, hidden and private in terms of significance. Maybe we should reconsider current description of 'significant' and declare it as flag important only for task_id generation.  If `significant` will be important only for task_id generation, then implementation of `Task.__repr__`  will be easier as it will be able to include only `public` parameters.

But currently, parameters significance could control its exposure in some cases

Reply:
Sorry for my absence here on this item.

I think for the matter of changing the definition of "significant parameters" and representation of a Task's `__repr`, we should ensure to get feedback from @honnix and @Tarrasch .

Reply:
I think what @nryanov said makes a lot of sense. However I think we've misused `__repr__` here. It should probably show *all* parameters. The issue however here is that we've totally misused it when we put it in the email header.

Maybe the email header could use `str()` instead of `repr()`. And we would implement

    def __str__() = return self._get_task_as_string(show_only_public_parameters=True)

 and

    def __repr__() = return self._get_task_as_string()

maybe.

Reply:
Thinking about it... using `str()` at that call site would also not be explicit, there we probably want to say we want it to be "give me string representation with PUBLIC params only' somehow.